cli/cmd/install/install.go

```diff
@@ -0,0 +1,500 @@
+package install
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/consul-k8s/cli/cmd/common"
+	"github.com/hashicorp/consul-k8s/cli/cmd/common/flag"
+	"github.com/hashicorp/consul-k8s/cli/cmd/common/terminal"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	helmCLI "helm.sh/helm/v3/pkg/cli"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	flagNamePreset = "preset"
+	defaultPreset  = ""
+
+	defaultReleaseName = "consul"
+
+	flagNameConfigFile      = "config-file"
+	flagNameSetStringValues = "set-string"
+	flagNameSetValues       = "set"
+	flagNameFileValues      = "set-file"
+
+	flagNameDryRun = "dry-run"
+	defaultDryRun  = false
+
+	flagNameAutoApprove = "auto-approve"
+	defaultAutoApprove  = false
+
+	flagNameNamespace = "namespace"
+	defaultNamespace  = "consul"
+
+	helmRepository = "https://helm.releases.hashicorp.com"
+)
+
+type Command struct {
+	*common.BaseCommand
+
+	kubernetes kubernetes.Interface
+
+	set *flag.Sets
+
+	flagPreset          string
+	flagNamespace       string
+	flagDryRun          bool
+	flagAutoApprove     bool
+	flagValueFiles      []string
+	flagSetStringValues []string
+	flagSetValues       []string
+	flagFileValues      []string
+
+	flagKubeConfig  string
+	flagKubeContext string
+
+	once sync.Once
+	help string
+}
+
+func (c *Command) init() {
+	// Store all the possible preset values in 'presetList'. Printed in the help message.
+	var presetList []string
+	for name := range presets {
+		presetList = append(presetList, name)
+	}
+
+	c.set = flag.NewSets()
+	{
+		f := c.set.NewSet("Command Options")
+		f.BoolVar(&flag.BoolVar{
+			Name:    flagNameAutoApprove,
+			Target:  &c.flagAutoApprove,
+			Default: defaultAutoApprove,
+			Usage:   "Skip confirmation prompt.",
+		})
+		f.BoolVar(&flag.BoolVar{
+			Name:    flagNameDryRun,
+			Target:  &c.flagDryRun,
+			Default: defaultDryRun,
+			Usage:   "Validate installation and return summary of installation.",
```

Comment:
This could be more explicit in telling the user that the dry run will not modify anything, maybe "Displays a summary of what would happen if installation were to run".

Reply:
I reworded to: Run pre-install checks and display summary of installation.